modules/rosa-sts-cluster-terraform.adoc

```diff
@@ -0,0 +1,547 @@
+  })
+}
+
+#Private link cluster Info
+variable "enable_private_link" {
+  type        = bool
+  description = "This enables private link"
+  default     = false
+}
+
+#VPC Info
+variable "create_vpc" {
+  type        = bool
+  description = "Create custom VPC for ROSA cluster"
+  default     = "false"
+}
+
+variable "vpc_name" {
+  type        = string
+  description = "VPC Name"
+  default     = "rosa-vpc"
+}
+
+variable "vpc_cidr_block" {
+  type        = string
+  description = "value of the CIDR block to use for the VPC"
+  default     = "10.0.0.0/16"
+}
+
+variable "private_subnet_cidrs" {
+  type        = list(any)
+  description = "The CIDR blocks to use for the private subnets"
+}
+
+variable "public_subnet_cidrs" {
+  type        = list(any)
+  description = "The CIDR blocks to use for the public subnets"
+  default     = ["10.0.101.0/24", "10.0.102.0/24", "10.0.103.0/24"]
+}
+
+variable "single_nat_gateway" {
+  type        = bool
+  description = "Single NAT or per NAT for subnet"
+  default     = false
+}
+
+#AWS Info
+variable "aws_region" {
+  type    = string
+  default = "us-east-1"
+}
+
+variable "availability_zones" {
+  type        = list(any)
+  description = "The availability zones to use for the cluster"
+  default     = ["us-east-1a"]
+}
+
+variable "installer_role_arn" {
+  description = "STS Role ARN with get secrets permission, relevant only for unmanaged OIDC config"
+  type        = string
+  default     = null
+}
+
+variable "private_subnet_ids" {
+  type        = list(any)
+  description = "VPC private subnets IDs for ROSA Cluster"
+  default     = []
+}
+EOF
+----
+
+.. Run the following command to create your `vpc.tf` file:
++
+[NOTE]
+====
+This guide does not use a VPC. You can change `create_vpc` to `true` in the `terraform.tfvars` file if you want to use a VPC.
+====
++
+[source,terminal]
+----
+$ cat<<-EOF>vpc.tf
+module "vpc" {
+  source  = "terraform-aws-modules/vpc/aws"
+  version = "5.1.2"
+
+  count = var.create_vpc ? 1 : 0
+  name  = var.vpc_name
+  cidr  = var.vpc_cidr_block
+
+  azs             = var.availability_zones
+  private_subnets = var.private_subnet_cidrs
+  public_subnets  = var.public_subnet_cidrs
+
+  enable_nat_gateway   = true
+  single_nat_gateway   = var.single_nat_gateway
+  enable_dns_hostnames = true
+  enable_dns_support   = true
+
+  tags = var.additional_tags
+}
+EOF
+----
+. Create a `terraform.tfvars` file by running the following command:
++
+[source,terminal]
+----
+$ cat<<-EOF>terraform.tfvars
+# Advanced ROSA Cluster Options
+worker_node_replicas = "2"
+
+#Private Link Cluster & VPC
+single_nat_gateway=true
+vpc_cidr_block="10.12.0.0/16"
+private_subnet_cidrs=["10.12.0.0/22", "10.12.4.0/22", "10.12.8.0/22"]
+public_subnet_cidrs=["10.12.128.0/24", "10.12.129.0/24", "10.12.130.0/24"]
+EOF
+----
+
+. In your terminal, run the following command in the directory where you saved these Terraform files to set up Terraform to create these resources:
```

Comment:
```suggestion
. In your terminal, run the following command in the directory where you saved the Terraform files to set up Terraform to create these resources:
```

Reply:
One too many "these" for my liking:) TBH, would like to come up with another way to phrase "to set up" as well but I am coming up with nothing :)